code/game/mecha/mech_fabricator.dm

```diff
@@ -88,6 +90,14 @@
 	QDEL_NULL(local_designs)
 	return ..()
 
+/obj/machinery/mecha_part_fabricator/multitool_act(mob/user, obj/item/I)
+	if(!panel_open)
+		return
+	if(!I.tool_start_check(src, user, 0))
+		return
+	output_dir = turn(output_dir, -90)
+	to_chat(user, "<span class='notice'>You change [src] to output to the [dir2text(output_dir)].</span>")
```

Comment:
Related to my other comment. Need to return true in tool acts to cancel the attackby/afterattack.
```suggestion
/obj/machinery/mecha_part_fabricator/multitool_act(mob/user, obj/item/I)
	if(!panel_open)
		return
	. = TRUE
	if(!I.tool_start_check(src, user, 0))
		return
	output_dir = turn(output_dir, -90)
	to_chat(user, "<span class='notice'>You change [src] to output to the [dir2text(output_dir)].</span>")
```

Reply:
ah, so doing this here means the other section can be avoided?

Reply:
I should have mentioned that, but yeah it should, because it cancels ever calling attackby. Could do with a test of course.

Reply:
this worked in testing, much better